safety-quiz/safetyQuiz.py

```diff
@@ -80,9 +88,71 @@ def index():
         quizzes = db.query(Machine).filter(Machine.quiz_id != None).order_by(Machine.quiz_id).all()
         return render_template('admin/index.html', trainings=trainings, quizzes=quizzes)
     else:
-        return render_template('index.html', trainings=trainings)
+        return render_template('welcome.html', trainings=trainings)
+
+@app.route('/new_trainings')
+def new_training_interface():
+    db = db_session()
+
+    trainings = db.query(Training).outerjoin(Machine).filter(Training.trainee_id == session['sid']).filter(\
+    Training.invalidation_date == None).filter(Machine.location_id.in_((2, 3))).order_by(Training.date).all()
+    quizzes = db.query(Machine).filter(Machine.quiz_id != None).order_by(Machine.quiz_id).all()
+    training_count = len(trainings)
+    return render_template('new_trainings.html', trainings=trainings, quizzes=quizzes, training_count = training_count)
+
+@app.route('/welcome')
+def welcome():
+    db = db_session()
+
+    user_count=db.query(Location.id,Location.name, Location.staff_ratio).all()
+
+    in_lab = [len(db.query(Access)\
+        .filter_by(timeOut=None) \
+        .filter_by(location_id=1)
+        .all()), len(db.query(Access)\
+        .filter_by(timeOut=None) \
+        .filter_by(location_id=2)
+        .all())]
+
 
 
+
+    return render_template('welcome.html', user_count = user_count, in_lab = in_lab)
+
+@app.route('/shop_status', methods=['GET','POST'])
+def shop_status():
+    db = db_session()
+    user_count = db.query(Location.id, Location.name, Location.staff_ratio).all()
+
+    in_lab = [len(db.query(Access) \
+                  .filter_by(timeOut=None) \
+                  .filter_by(location_id=1)
+                  .all()), len(db.query(Access) \
+                  .filter_by(timeOut=None) \
+                  .filter_by(location_id=2)
+                  .all())]
+
+    if request.method =='GET':
+        return render_template('shop_status.html', user_count=user_count, in_lab=in_lab)
+    elif request.method == 'POST':
+        response = app.make_response('<h1>Not yet implemented!</h1>'), 418
+        return response
+
+@app.route('/COVID',methods=['GET', 'POST'])
```

Comment:
This function and route has been moved outside of safetyQuiz.py

Reply:
Still needs to be resolved